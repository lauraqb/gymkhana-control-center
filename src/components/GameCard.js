import React from 'react'
import { Link } from 'react-router-dom'
import Card from 'react-bootstrap/Card'
import { IconContext } from "react-icons"
import { GiAbstract118 } from "react-icons/gi"
import './styles/GameCard.css'

export class GameCard extends React.Component {
    constructor(props) {
      super(props)
      this.nombrePartida = this.props.nombre
      this.pin = this.props.pin
      this.link = "./game/"+this.props.id+"?clave="+this.pin
      this.linkChallenges = "./game/"+this.props.id+"/challenges"
    }

    render() {
        return  <Link to={this.link} className="g-link">
        <Card className="g-card">
          <Card.Body>
          <div className="icono">
              <IconContext.Provider value={{ color: "#5a80a9", size:	"2.50em" }}>
                  <GiAbstract118 />
              </IconContext.Provider>
            </div>
            <div>
              <Card.Text className="g-card-title">{this.nombrePartida}</Card.Text>
              <Card.Text>PIN: {this.pin}</Card.Text>
              <Link to={this.linkChallenges} className="g-link">
              <Card.Link href="#">Pruebas JSON</Card.Link>
              </Link>
            </div>
          </Card.Body>
        </Card>
      </Link>
    }
}

export default GameCard