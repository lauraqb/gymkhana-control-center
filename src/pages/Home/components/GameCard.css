.g-card .card-body{
    display: flex;
    flex-direction: row;
}

.g-card .form-control, .g-card .btn {
    font-size: 1em;
}

.g-teams-container {
    margin-left: 15px;
}

.g-card p {
    margin-bottom: 0;
}

.g-card .card-body {
    padding: 1.0rem;
    /* cursor: pointer; */
}

.g-card-title {
    font-weight: bold;
}

.g-card:hover {
    background-color: aliceblue;
}

.g-card .btn-link {
    padding: 0
}