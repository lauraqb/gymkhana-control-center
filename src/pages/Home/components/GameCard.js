import React from 'react'
import { Link } from 'react-router-dom'
import axios from 'axios'
import Card from 'react-bootstrap/Card'
import Button from 'react-bootstrap/Button'
import Form from 'react-bootstrap/Form'
import Col from 'react-bootstrap/Col'

import Accordion from 'react-bootstrap/Accordion'
import { IconContext } from "react-icons"
import { GiAbstract118 } from "react-icons/gi"
import './GameCard.css'

const endpoint = process.env.REACT_APP_SERVER_ENDPOINT

export class GameCard extends React.Component {
    constructor(props) {
      super(props)
      this.state = {
        teams: null,
        teamName: "",
        teamKey: ""
      }
      
      this.id = this.props.id
      this.nombrePartida = this.props.nombre
      this.pin = this.props.pin
      this.link = "./game/"+this.props.id+"?clave="+this.pin
      this.linkChallenges = "./game/"+this.props.id+"/challenges"
      this.handleChange = this.handleChange.bind(this)
      this.getTeams()
      this.addNewTeam = this.addNewTeam.bind(this)
    }

    
    handleChange(event) {
        const value = event.target.value
        const name = event.target.name
        this.setState({
          [name]: value,
        })
    }

    getTeams = () => {
      axios.get(endpoint+"/cc/games/"+this.props.id+"/teams")
      .then(res => {
          this.setState({
              teams: res.data
          })
      })
      .catch(error => this.setState({ error: error.message }))
    }

    addNewTeam = (e) => {
      e.preventDefault()
      axios.put(endpoint+"/cc/games/"+this.props.id+"/teams", {teamName: this.state.teamName, teamKey: this.state.teamKey})
      .then(res => {
        this.getTeams()
      })
      .catch(error => console.error(error.message))
    }

    render() {
      if(this.state.error) {
        return <div>{this.state.error}</div>
      }
      return (
          <Card className="g-card" >
            <Card.Body>
              <Link to={this.link} className="g-link">
                <div className="icono">
                  <IconContext.Provider value={{ color: "#5a80a9", size:	"2.50em" }}> <GiAbstract118 /> </IconContext.Provider>
                </div>
              </Link>
              <div>
                  <Link to={this.link} className="g-link">
                    <Card.Text className="g-card-title">{this.nombrePartida}</Card.Text>
                  </Link>
                  <Card.Text>PIN: {this.pin}</Card.Text>
                  <Link to={this.linkChallenges} className="g-link">
                    <Button variant="link">Pruebas JSON</Button>
                  </Link>
                  <Accordion defaultActiveKey="0">
                      <Accordion.Toggle as={Button} variant="link" eventKey="1" data-test="g-expand-teams-btn">
                          Equipos >
                      </Accordion.Toggle>
                      <Accordion.Collapse eventKey="1">
                          <div className="g-teams-container">
                            {this.state.teams && this.state.teams.length === 0 && <span> No hay ningún equipo creado</span>}
                            {this.state.teams && this.state.teams.length > 0 && <
                              div className="g-teamslist" data-test="teamslist">{this.state.teams.map((team, index) => <div key={index}>- {team.name}, <span>{team.key}</span> </div>)}</div>
                              }

                            <Form onSubmit={this.addNewTeam}>
                              <Form.Row>
                                  <Col><Form.Control placeholder="Nombre equipo" name="teamName" value={this.state.teamName} onChange={this.handleChange}/></Col>
                                  <Col><Form.Control placeholder="Clave" name="teamKey" value={this.state.teamKey} onChange={this.handleChange}/></Col>
                                  <Button variant="primary" type="submit">Añadir nuevo equipo</Button>
                              </Form.Row>
                            </Form>
                          </div>
                      </Accordion.Collapse>
                  </Accordion>


                
              
              </div>
            </Card.Body>
          </Card>
      
      )
    }
}

export default GameCard