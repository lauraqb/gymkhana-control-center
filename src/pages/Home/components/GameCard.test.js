import React from 'react'
import { shallow } from 'enzyme'
import GameCard from './GameCard'

let component

beforeEach(() => component = shallow(<GameCard/>))

describe('GameCard component: ', () => {
    it ('Renders correctly', () => {
        expect(component).toMatchSnapshot()
    })
    it ('Does not exist list of teams in div if state.teams has no data', () => {
        expect(component.find('[data-test="teamslist"]')).toHaveLength(0)
    })
    it ('Loads list of teams in div if state.teams has data', () => {
        component.setState({teams: [{name: "test", key:"123"}]})
        expect(component.find('[data-test="teamslist"]')).toHaveLength(1)
    })
    it ('calls function after add new team', () => {
        const button = component.find('[data-test="g-expand-teams-btn"]')
        expect(button).toHaveLength(1)
        button.simulate('click')
        //TODO: testear que funcion es llamada
    })
})