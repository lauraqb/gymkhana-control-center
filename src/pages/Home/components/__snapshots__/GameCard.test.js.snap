// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard component:  Renders correctly 1`] = `
<Card
  body={false}
  className="g-card"
>
  <CardBody>
    <Link
      className="g-link"
      to="./game/undefined?clave=undefined"
    >
      <div
        className="icono"
      >
        <ContextProvider
          value={
            Object {
              "color": "#5a80a9",
              "size": "2.50em",
            }
          }
        >
           
          <GiAbstract118 />
           
        </ContextProvider>
      </div>
    </Link>
    <div>
      <Link
        className="g-link"
        to="./game/undefined?clave=undefined"
      >
        <CardText
          className="g-card-title"
        />
      </Link>
      <CardText>
        PIN: 
      </CardText>
      <Link
        className="g-link"
        to="./game/undefined/challenges"
      >
        <Button
          active={false}
          disabled={false}
          type="button"
          variant="link"
        >
          Pruebas JSON
        </Button>
      </Link>
      <ForwardRef
        defaultActiveKey="0"
      >
        <ForwardRef
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            }
          }
          data-test="g-expand-teams-btn"
          eventKey="1"
          variant="link"
        >
          Equipos &gt;
        </ForwardRef>
        <AccordionCollapse
          eventKey="1"
        >
          <div
            className="g-teams-container"
          >
            <Form
              inline={false}
              onSubmit={[Function]}
            >
              <FormRow>
                <Col>
                  <FormControl
                    name="teamName"
                    onChange={[Function]}
                    placeholder="Nombre equipo"
                    value=""
                  />
                </Col>
                <Col>
                  <FormControl
                    name="teamKey"
                    onChange={[Function]}
                    placeholder="Clave"
                    value=""
                  />
                </Col>
                <Button
                  active={false}
                  disabled={false}
                  type="submit"
                  variant="primary"
                >
                  AÃ±adir nuevo equipo
                </Button>
              </FormRow>
            </Form>
          </div>
        </AccordionCollapse>
      </ForwardRef>
    </div>
  </CardBody>
</Card>
`;
